### PR TITLE
MTDSA-14676 v4 MTD endpoint for List BF Losses

### PR DESCRIPTION
**Pull request by mattstephens-hmrc**

### CODE REVIEW

File: app/api/endpoints/bfLoss/connector/v3/BFLossConnector.scala [hunk 1]
```diff
@@ -72,4 +81,47 @@ class BFLossConnector @Inject() (val http: HttpClient, val appConfig: AppConfig)
     get(IfsUri[RetrieveBFLossResponse](s"income-tax/brought-forward-losses/$nino/$lossId"))
   }
 
+  def listBFLosses(
```
Comment (tom-lloyd-hmrc) [inline app/api/endpoints/bfLoss/connector/v3/BFLossConnector.scala:84]: Why has this been added back here? List BF Losses has its own connector.

File: it/api/endpoints/bfLoss/list/v3/ListBFLossesControllerISpec.scala [hunk 2]
```diff
@@ -103,57 +49,15 @@ class ListBFLossesControllerISpec extends V3IntegrationBaseSpec {
         response.header("Content-Type") shouldBe Some("application/json")
       }
 
-      "query without a tax year" in new Test {
```
Comment (tom-lloyd-hmrc) [inline it/api/endpoints/bfLoss/list/v3/ListBFLossesControllerISpec.scala:106]: Has this file been the victim of a bad merge? Looks to be reverting all the changes made for the V3 changes to make multiple downstream requests.

File: test/api/endpoints/bfLoss/list/v4/connector/ListBFLossesConnectorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+          willGet(
+            url = s"$baseUrl/income-tax/brought-forward-losses/${taxYear.asTysDownstream}/$nino"
+          ).returns(Future.successful(downstreamResponse))
+
+          val result: DownstreamOutcome[ListBFLossesResponse[ListBFLossesItem]] = await(connector.listBFLosses(request))
+          result shouldBe downstreamResponse
+        }
+
+        "a valid non-TYS request with all parameters" in new TysIfsTest with Test {
```
Comment (tom-lloyd-hmrc) [inline test/api/endpoints/bfLoss/list/v4/connector/ListBFLossesConnectorSpec.scala:71]: Don't think it makes sense to call this non-TYS as there's no distinction any more, also I think it's using the 2023-24 tax year anyway!

Comment (mattstephens-hmrc) [reply test/api/endpoints/bfLoss/list/v4/connector/ListBFLossesConnectorSpec.scala:71]: good spot, thought I'd deleted all the occurrences

**Review by tom-lloyd-hmrc — APPROVED** [review]

**Pull request merged**